code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -173,7 +173,7 @@
 	amount_per_transfer_from_this = 10
 	possible_transfer_amounts = list(5,10,15,20,25,30,60,120)
 
-/obj/item/reagent_containers/glass/beaker/plastic/update_icon_state()
+/obj/item/reagent_containers/glass/beaker/plastic/update_overlays()
```

Comment:
This should be on ``update_icon`` instead i think, this isn't changing overlays

Reply:
it is, it's calling the super `/obj/item/reagent_containers/glass/beaker/update_overlays()` which is responsible for actually updating overlays. this function temporarily overrides the icon_state but then sets it back to its normal value, so after the function call is done the state has not actually changed.

Reply:
That proc is actually adding an overlay onto the beaker, this is not, so it should be on ``update_icon``, the proc that calls ``update_overlays``.

https://github.com/yogstation13/Yogstation/assets/53777086/f51ca596-b485-496c-990d-45197bf2c59d